app/controllers/api/v1/reservations_controller.rb

```diff
@@ -0,0 +1,53 @@
+class Api::V1::ReservationsController < ApplicationController
+  before_action :set_reservation, only: %i[show update destroy]
+
+  # GET /reservations
+  def index
+    user = User.find(params[:user_id])
+    @reservations = user.reservations.includes(:car).where(cars: { is_removed: false })
+
+    render json: @reservations, include: :car
+  end
+
+  # GET /reservations/1
+  def show
+    render json: @reservation
+  end
+
+  # POST /reservations
+  def create
+    @reservation = Reservation.new(reservation_params)
+
+    if @reservation.save
+      render json: @reservation, status: :created
+    else
+      render json: @reservation.errors, status: :unprocessable_entity
+    end
+  end
+
+  # PATCH/PUT /reservations/1
+  def update
+    if @reservation.update(reservation_params)
+      render json: @reservation
+    else
+      render json: @reservation.errors, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @reservation.destroy
+    head :no_content
+  end
```

Comment:
- [ ] What happens here if the reservation is not successfully destroyed?  Kindly, add some error handling just like you did in the update method (line 29). 👍